File: app/channels/post_channel.rb
class PostChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from "post_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    ActionCable.server.broadcast 'post_channel', comment: data['comment']
  end
end

File: app/javascript/channels/post_channel.js
import consumer from "./consumer"

const appRoom = consumer.subscriptions.create("PostChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    return alert(data['comment']);
  },

  speak: function(comment) {
    return this.perform('speak', {comment: comment});
  }

});

window.addEventListener("keypress", function(e) {
  if (e.key === "Enter") {
    appRoom.speak(e.target.value);
    e.target.value = '';
    e.preventDefault();
  }
})